pkg/client/vault.go

```diff
@@ -186,6 +188,134 @@ func (api *APIClient) UpdateSecretFile(vault, secret, eTag string, content []byt
+	AffiliationName string        `json:"affiliationName"`
+	Permissions     []string      `json:"permissions"`
+	Secrets         []SecretInput `json:"secrets"`
+	VaultName       string        `json:"vaultName"`
+}
+
+// CreateVaultInput is input to the graphql createVault interface
+type SecretInput struct {
+	Base64Content string `json:"base64Content"`
+	Name          string `json:"name"`
+}
+
+// AddPermissions adds permissions to vault via gobo
+func (api *APIClient) CreateVault(vault AuroraSecretVault) error {
+	if len(vault.Secrets) == 0 {
+		return errors.New(FoundNoSecretsForVault)
+	}
+
+	createVaultRequest := graphql.NewRequest(createVaultRequestString)
+	createVaultInput := getCreateVaultInput(vault, api.Affiliation)
+	createVaultRequest.Var("createVaultInput", createVaultInput)
+	var createVaultResponse VaultResponse
+	if err := api.RunGraphQlMutation(createVaultRequest, &createVaultResponse); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getCreateVaultInput(vault AuroraSecretVault, affiliation string) CreateVaultInput {
+	secrets := make([]SecretInput, len(vault.Secrets))
+	i := 0
+	for key, content := range vault.Secrets {
+		secrets[i] = SecretInput{
+			Base64Content: content,
+			Name:          key,
+		}
+		i++
+	}
+	createVaultInput := CreateVaultInput{
+		AffiliationName: affiliation,
+		Permissions:     vault.Permissions,
+		VaultName:       vault.Name,
+		Secrets:         secrets,
+	}
+	return createVaultInput
+}
+
+const addVaultPermissionsRequestString = `mutation addVaultPermissions($addVaultPermissionsInput: AddVaultPermissionsInput!){
+  addVaultPermissions(input: $addVaultPermissionsInput)
+  {
+    hasAccess
+    name
+    permissions
+  }
+}`
+
+// VaultPermissionsInput is input to the graphql addVaultPermissions and removeVAultPermissions interfaces
+type VaultPermissionsInput struct {
+	AffiliationName string   `json:"affiliationName"`
+	Permissions     []string `json:"permissions"`
+	VaultName       string   `json:"vaultName"`
+}
+
+// AddPermissions adds permissions to vault via gobo
+func (api *APIClient) AddPermissions(vaultName string, permissions []string) error {
+	addVaultPermissionsRequest := graphql.NewRequest(addVaultPermissionsRequestString)
+	addVaultPermissionsInput := VaultPermissionsInput{
+		AffiliationName: api.Affiliation,
+		Permissions:     permissions,
+		VaultName:       vaultName,
+	}
+	addVaultPermissionsRequest.Var("addVaultPermissionsInput", addVaultPermissionsInput)
+
+	var addVaultPermissionsResponse VaultResponse
+	if err := api.RunGraphQlMutation(addVaultPermissionsRequest, &addVaultPermissionsResponse); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+const removeVaultPermissionsRequestString = `mutation removeVaultPermissions($removeVaultPermissionsInput: RemoveVaultPermissionsInput!){
+  removeVaultPermissions(input: $removeVaultPermissionsInput)
+  {
+    hasAccess
+    name
+    permissions
+  }
+}`
+
+// RemovePermissions removes permissions from vault via gobo
```

Comment:
Var det sånn at public metoder i go må ha kommentar?
litt uvant for meg, og denne kommentaren nevner implementasjonsdetaljer som kanskje er litt unødvendig?

Reply:
Jeg har innført en ganske standard Makefile med linting tidligere, og da får man ganske strenge krav til dette å forholde seg til.  Det henger igjen.  Du må ikke ha kommentarer, men med linting så er dette det anbefalte.